File: features/master/src/main/java/com/nivasafinance/features/master/codemaster/exception/CodeMasterOperationException.java
package com.nivasafinance.features.master.codemaster.exception;

import com.nivasafinance.common.exception.ExceptionUtils;
import org.springframework.context.MessageSource;

import java.io.Serial;

public class CodeMasterOperationException extends RuntimeException {
    
    @Serial
    private static final long serialVersionUID = 1L;
    
    public CodeMasterOperationException(String messageKey, MessageSource messageSource) {
        super(ExceptionUtils.createLocalizedMessage(messageKey, null, messageSource));
    }
}

File: features/master/src/main/java/com/nivasafinance/features/master/codemaster/exception/CodeMasterOperationException.kt
package com.nivasafinance.features.master.codemaster.exception

import com.nivasafinance.common.exception.ExceptionUtils
import org.springframework.context.MessageSource

class CodeMasterOperationException(
    messageKey: String,
    messageSource: MessageSource
) : RuntimeException(
    ExceptionUtils.createLocalizedMessage(messageKey, null, messageSource)
)
